File: core/src/test/resources/java-samples/SampleCoupling.java
package com.example;

import com.example.domain.User;

import java.util.ArrayList;
import java.util.List;

public class SampleCoupling {
    private User user; // Internal Coupling (+1)
    private List<String> names = new ArrayList<>(); // External coupling (List and ArrayList)

    public void processContent(User u) { // Internal Coupling (+1)
        String s = "test"; // External coupling (String)
    }
}

File: core/src/test/resources/kotlin-samples/com/examples/SampleCoupling.kt
package com.example

import com.example.domain.*

class SampleCoupling {
    fun testCoupling() {
        val user = User("John") // +1 Internal Coupling
        InternalClass().hello() // +1 Internal Coupling
        House().hello() // +1 Internal Coupling
        println(user.name)
    }
}


internal class InternalClass {
    fun hello() {

    }
}